### PR TITLE
CP-23095 Inject IGMP snooping query message when VM migrate completes

### PR DESCRIPTION
**Pull request by krizex**
Signed-off-by: Yang Qian <yang.qian@citrix.com>

Partial done, create PR for internal review.

### CODE REVIEW

File: lib/xenops_server.ml [hunk 1]
```diff
@@ -1100,6 +1101,31 @@ let rec perform_atomic ~progress_callback ?subtask ?result (op: atomic) (t: Xeno
 			debug "VIF.set_active %s %b" (VIF_DB.string_of_id id) b;
 			B.VIF.set_active t (VIF_DB.vm_of id) (VIF_DB.read_exn id) b;
 			VIF_DB.signal id
+        | VIF_inject_igmp_query id ->
```
Comment (huizh) [inline lib/xenops_server.ml]: Consider to add a common function to wait for VIF active.

File: lib/xenops_server.ml [hunk 2]
```diff
@@ -1100,6 +1101,31 @@ let rec perform_atomic ~progress_callback ?subtask ?result (op: atomic) (t: Xeno
...
+                  true
+                end else
+                begin
+                  debug "vif %s status change to '%s'" (VIF_DB.string_of_id id) status;
+                  false
+                end
+            | _ -> false
+          in
+          (* wait until vif reconnected of timeout *)
+          B.UPDATES.event_wait 10.0 f |> ignore;
```
Comment (huizh) [inline lib/xenops_server.ml]: Consider a shorter timeout and talk with ring3 expert for suggestion.

**Comment by huizh** [comment]
General comment: update the indents to make them consistent.

**Comment by robhoes** [comment]
We have discussed a different way of doing this, so I'm closing the PR for now.

**Pull request closed without merging**